Source/MQTTnet/MqttApplicationMessageReceivedEventArgs.cs

```diff
@@ -29,5 +30,7 @@ public MqttApplicationMessageReceivedEventArgs(string clientId, MqttApplicationM
         public bool IsHandled { get; set; }
         
         public object Tag { get; set; }
+
+        public Task<bool> PendingTask { get; set; }
```

Comment:
I checked how other libraries are handling this stuff and I would like go another way. When using the Service Bus SDK for Microsoft Azure you can do basically the same but there you have an "AcknowledgeMode" property with enum values "Automatic | Manual". The default is "Automatic" and the handler has to change this to "Manual" for every message where the caller will finalize. Additionally the handler context has a method called "Acknowledge()". This can be called at any time from any thread later. In our case the ACK packet for QoS1 will be sent then. For QoS0 nothing will happen (also no exception). In my opinion this approach looks much cleaner than having a floating Task propery which must be assigned etc.

Reply:
I agree that it's better to model the solution after a more widespread pattern - my solution with `PendingTask` fits better to the code I'm working on, but I can adapt that easily. I've changed the implementation as requested; I've also updated the description above to match the current implementation.